### PR TITLE
Fix a crash on restore down

### LINKED ISSUE #1795: BUG: Minimizing Terminal Crash
# Environment

```none
Windows build number: 10.0.18362.175
Windows Terminal version: 0.2.1831.0
```

# Steps to reproduce
Execute a command that keeps outputting something for a while.
for example you can user `apt list` or `ls` in a directory with a lot of content.

Then, while the command is running, double click on the titlebar or click the minimize? button.

# Expected behavior
The behavior I would have expected was for it to pause or keep running
# Actual behavior
The Terminal Freezes and crashes after approximately 1,5 seconds
Other Terminal windows are unaffected

### PR DESCRIPTION
**Pull request by zadjii-msft**
## Summary of the Pull Request

Fixes a crash that occur when changing the buffer height. During the resize, conpty would immediately request a repaint, which would throw an exception that would cause the resize to fail halfway through, and that would make the subsequent render fail.

This change prevents the VT Renderer from immediately requesting a paint during the resize. 

 
## References

Might also be related to #1095. I can't seem to get that to repro anymore with this change.

Might be related to #1856. We should verify.

This does however make another issue more apparent. When you restore down with a COOKED_READ input line (read:cmd.exe), the terminal can crash. I have a fix for that in d34a3be, which I wasn't sure if I should include in this PR or not, because I wasn't confident that was the right fix. I'm fairly certain there's another issue floating around tracking that bug. We could either merge that commit/branch into this one, or wait and save it for a separate one. 

## PR Checklist
* [x] Closes #1795

### CODE REVIEW

**Comment by miniksa** [comment]
For your tentative COOKED_READ fix, that is sort of how the v1 console used to do it when resizing anything. It would hide the cooked line, do the work, then redraw it in the new position. So I'm not really opposed to that.

My thought would be... should it belong to `SCREEN_INFORMATION::SetViewportSize` instead, though, because I bet other folks resizing it probably also have the same problem if it's true here.

File: src/host/VtIo.cpp [hunk 1]
```diff
@@ -397,3 +397,37 @@ void VtIo::_ShutdownIfNeeded()
...
+// - Tell the vt renderer to begin a resize operation. During a resize
+//   operation, the vt renderer should _not_ request to be repainted during a
+//   text buffer circling event. Any callers of this method should make sure to
+//   call EndResize to make sure the renderer returns to normal behavior.
+//   See GH#1795 for context on this method.
+// Arguments:
+// - <none>
+// Return Value:
+// - <none>
+void VtIo::BeginResize()
```
Comment (miniksa) [inline src/host/VtIo.cpp]: Uh, maybe this method should just return an RAII object that stops the resize when it is destroyed like what we envision good locks to do.

Comment (zadjii-msft) [reply src/host/VtIo.cpp]: You know, I almost wrote it that way. I'll fix it.

**Comment by DHowett-MSFT** [comment]
Is this something we should cherry-pick into 20H1?

**Comment by zadjii-msft** [comment]
@DHowett-MSFT Probably. 

@miniksa Okay that makes sense. I'll make another PR for that one, since this one has a pretty specific scope, as does the other.

**Comment by zadjii-msft** [comment]
@DHowett-MSFT Oh you said 20H1, not Vb. Eh, feels too risky for 20H1.

File: src/host/VtIo.cpp (hunk 1 above)
Comment (zadjii-msft) [reply src/host/VtIo.cpp]: Moved to #2162

**Comment by zadjii-msft** [comment]
I was wrong this should be in 20H1 (which is Vb). Our codenames are confusing.

**Pull request merged**